src/NHibernate/Proxy/ProxyBuilderHelper.cs

```diff
@@ -101,11 +102,12 @@ internal static IEnumerable<MethodInfo> GetProxiableMethods(System.Type type)
 
 		internal static IEnumerable<MethodInfo> GetProxiableMethods(System.Type type, IEnumerable<System.Type> interfaces)
 		{
-			var proxiableMethods =
-				GetProxiableMethods(type)
-					.Concat(interfaces.SelectMany(i => i.GetMethods()))
-					.Distinct();
-			
+			var proxiableMethods = new HashSet<MethodInfo>(GetProxiableMethods(type), new MethodInfoComparer(type));
+			foreach (var interfaceMethod in interfaces.SelectMany(i => i.GetMethods()))
+			{
+				proxiableMethods.Add(interfaceMethod);
+			}
```

Comment:
I could have kept the previous code, just giving the comparer to `Distinct`. But its contract does not guarantee we will takes the base type methods rather than "duplicated" methods from the interfaces. (Its current implementation will keep the methods from the base type, but we should not rely on this.)
By explicitly adding the base methods to the hashset first, it is guaranteed that this is dups from interfaces which will be ignored.